deps/wrapper.cc

```diff
@@ -149,9 +149,11 @@ ObjectID _submit_task(const ray::JuliaFunctionDescriptor &jl_func_descriptor,
     ray::FunctionDescriptor func_descriptor = std::make_shared<ray::JuliaFunctionDescriptor>(jl_func_descriptor);
     RayFunction func(Language::JULIA, func_descriptor);
 
+    // TODO: Passing in a `std::vector<std::unique_ptr<TaskArg>>` from Julia may currently be impossible due to:
+    // https://github.com/JuliaInterop/CxxWrap.jl/issues/370
     std::vector<std::unique_ptr<TaskArg>> args;
-    for (auto & obj_id : object_ids) {
-        args.emplace_back(new TaskArgByReference(obj_id, worker.GetRpcAddress(), /*call-site*/""));
+    for (auto &task_arg : task_args) {
+        args.emplace_back(task_arg);
```

Comment:
since the element type is `unique_ptr` already I'm assuming this will auto-cast to `unique_ptr`?

Reply:
ah wait I always forget, `emplace_back` constructs and pushes...

Reply:
> since the element type is `unique_ptr` already I'm assuming this will auto-cast to `unique_ptr`?

Just to avoid any possible confusion the `task_arg` element is just a raw pointer and is not already a `unique_ptr`